dali/plugin/plugin_manager.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dlfcn.h>
+#include "dali/plugin/plugin_manager.h"
+#include "dali/error_handling.h"
+
+namespace dali {
+
+PluginManager& PluginManager::Instance() {
+    static PluginManager instance;
+    return instance;
+}
+
+PluginManager::~PluginManager() {
+    // WARNING! Calling dlclose will produce a crash because the lifecycle
+    //          of SchemaRegistry and OperatorRegistry is not managed
+    // TODO(janton): consider doing dlclose after refactoring of the registry's lifecycle
```

Comment:
I think we will never want to call the `dlclose` here, since the only moment when we want it to close it is when DALI exits (and that's automatically done by the system)

Reply:
In that case I can simplify the PluginManager by removing the destructor and the std::vector<void*>